### PR TITLE
VisitorHelper Addition

### PR DESCRIPTION
**Pull request by UpFault**
onMouseClicked in VisitorHelper.java now copies the item amount to clipboard

### CODE REVIEW

**Comment by kevinthegreat1** [comment]
Would be nice to have a text saying `[Click to Copy Amount]` or `[Copy Amount]`

**Comment by UpFault** [comment]
rather than just copying it, yeah I can definitely make that happen.

File: src/main/java/de/hysky/skyblocker/skyblock/garden/VisitorHelper.java [hunk 1]
```diff
@@ -69,19 +69,25 @@ public static void onMouseClicked(double mouseX, double mouseY, int mouseButton,
...
+                int copyAmountTextX = TEXT_START_X + visitorNameWidth + 7;
 
                 if (isMouseOverText(mouseX, mouseY, TEXT_START_X + TEXT_INDENT, yPosition, textWidth, textHeight)) {
                     MessageScheduler.INSTANCE.sendMessageAfterCooldown("/bz " + itemText);
+                }
 
+                if (isMouseOverText(mouseX, mouseY, copyAmountTextX, yPosition, textRenderer.getWidth("[Copy Amount]"), textHeight)) {
                     MinecraftClient.getInstance().keyboard.setClipboard(String.valueOf(itemEntry.getIntValue()));
                     return;
                 }
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/garden/VisitorHelper.java:85]: Change the x to not overlap and add some feedback to the player. You will have to import some stuff.
```suggestion
                MinecraftClient client = MinecraftClient.getInstance();
                if (client.player != null && isMouseOverText(mouseX, mouseY, TEXT_START_X + TEXT_INDENT + textWidth, yPosition, textRenderer.getWidth(" [Copy Amount]"), textHeight)) {
                    client.keyboard.setClipboard(String.valueOf(itemEntry.getIntValue()));
                    client.player.sendMessage(Constants.PREFIX.get().append("Copied amount successfully"));
                    return;
                }
```

File: src/main/java/de/hysky/skyblocker/skyblock/garden/VisitorHelper.java [hunk 2]
```diff
@@ -135,7 +141,9 @@ private static void drawScreenItems(DrawContext context, TextRenderer textRender
         int index = 0;
         for (Map.Entry<Pair<String, String>, Object2IntMap<String>> visitorEntry : itemMap.entrySet()) {
             Pair<String, String> visitorName = visitorEntry.getKey();
+            Text visitorNameText = Text.literal(visitorName.left());
             drawTextWithOptionalUnderline(context, textRenderer, Text.literal(visitorName.left()), TEXT_START_X, TEXT_START_Y + index * (LINE_SPACING + textRenderer.fontHeight), mouseX, mouseY);
+            drawTextWithOptionalUnderline(context, textRenderer, Text.literal("[Copy Amount]"), textRenderer.getWidth(visitorNameText) + 7, TEXT_START_Y + index * (LINE_SPACING + textRenderer.fontHeight), mouseX, mouseY);
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/garden/VisitorHelper.java:146]:
```suggestion
            drawTextWithOptionalUnderline(context, textRenderer, Text.literal(visitorName.left() + " [Copy Amount]"), TEXT_START_X, TEXT_START_Y + index * (LINE_SPACING + textRenderer.fontHeight), mouseX, mouseY);
```

File: src/main/java/de/hysky/skyblocker/skyblock/garden/VisitorHelper.java [hunk 3]
```diff
@@ -69,19 +69,25 @@ public static void onMouseClicked(double mouseX, double mouseY, int mouseButton,
...
 
+                if (isMouseOverText(mouseX, mouseY, copyAmountTextX, yPosition, textRenderer.getWidth("[Copy Amount]"), textHeight)) {
                     MinecraftClient.getInstance().keyboard.setClipboard(String.valueOf(itemEntry.getIntValue()));
                     return;
                 }
+
                 yPosition += LINE_SPACING + textHeight;
             }
         }
     }
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/garden/VisitorHelper.java:90]:
```suggestion
    }

```

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]
Thanks for your work so far. Once you address these it should be good to go!

File: src/main/java/de/hysky/skyblocker/skyblock/garden/VisitorHelper.java (hunk 1 above)
Comment (UpFault) [reply src/main/java/de/hysky/skyblocker/skyblock/garden/VisitorHelper.java]: gotta fix the clipboard, doesn't wanna copy with this change :/

**Comment by UpFault** [comment]
jesus, was supposed to be 1 final commit, anyways, took the changes you suggested, added them and fixed the yPos issue I was having which is why it wasn't sending the player feedback or copying to the clipboard, everything should be good now.

**Review by UpFault — COMMENTED** [review]
everything looks good

**Pull request closed without merging**